### ISSUE TITLE
[Global] Pop Up Block's Modal container does not receive focus

### ISSUE THREAD

**Issue by nicklepine** [Shopify/dawn#80]
## Issue summary

When a pop up modal is activated, focus is sent to the web component container. Screen reader users may not be made aware of the modal context presented to them as this container does not feature the required semantic meaning.

- macOS VoiceOver does not announce any context change
- iOS VoiceOver shifts focus back to the top of the DOM
- JAWS does not announce the dialog context as focus shifts to the container outside of this context

This would result in a confusing or frustrating user experience for screen reader users.

<details>
  <summary>Screenshots</summary>

![Animated gif showcasing Chrome devtools outputting currently focused DOM node. When modal window appears, focus shifts to parent container.](https://user-images.githubusercontent.com/1392632/116922905-f8076880-ac23-11eb-9c08-76a9dd696d33.gif)

</details>

#### HTML

```html
<product-modal class="product-media-modal" tabindex="-1" data-index="1" open="">
  <div role="dialog" aria-label="Media gallery" aria-modal="true">
    <!-- … -->
  </div>
</product-modal>
```

## Steps to reproduce

1. Go to a product page that contains a pop up block with any supported browser
2. With and active screen reader, navigate to the pop up modal window activator control
3. Activate one of the controls

### Expected

- Modal window to receive focus and the announcement of "dialog" to share context

### Actual

- Parent container receives focus instead, resulting in loss of context

## Recommendation

1. When the control is activated, shift focus to the `role="dialog"` container via `focus()` method. This will allow for the context state to be conveyed correctly.
2. Move the `tabindex` attribute from the parent container onto the `role="dialog"` container in order for it to receive focus.

#### HTML

```html
<product-modal class="product-media-modal" data-index="1" open="">
  <div role="dialog" aria-label="Media gallery" aria-modal="true" tabindex="-1">
    <!-- … -->
  </div>
</product-modal>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **WCAG Principal:** Perceivable
- **WCAG SC:** [1.3.1: Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
- **Severity:** High
- **Effort:** Medium
cc: @svinkle
